lib/src/react_client/dart_interop_statics.dart

```diff
@@ -33,8 +33,9 @@ final ReactDartInteropStatics dartInteropStatics = (() {
           var ref = getProperty(jsThis.refs, name);
           if (ref == null) return null;
           if (ref is Element) return ref;
+          if (ref is ReactComponent) return ref.dartComponent ?? ref;
 
-          return (ref as ReactComponent).dartComponent ?? ref;
+          return ref;
```

Comment:
These changes were to fix an error when setting a string ref on a component using `useImperativeHandle` (because it would try to cast the customized ref to `ReactComponent`).

This an edge-case, but it was easier to fix the issue than to skip testing that case.